src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -208,18 +195,14 @@ public Dataset retrieveByName(String name) {
    * object. Unlike the Dataset object, the Model object includes a reference to the associated
    * cloud project.
    *
-   * <p>Note that this method will only return a dataset if it is NOT exclusively locked. It is
-   * intended for user-facing calls (e.g. from RepositoryApiController), not internal calls that may
-   * require an exclusively locked dataset to be returned (e.g. dataset deletion).
-   *
    * @param id in UUID format
    * @return a DatasetModel = API output-friendly representation of the Dataset
    */
-  public DatasetModel retrieveAvailableDatasetModel(
+  public DatasetModel retrieveDatasetModel(
```

Comment:
With this change you could change `DeleteDatasetAuthzBqAclsStep.doStep()` to call this instead of
```
      Dataset dataset = datasetService.retrieve(datasetId);
      datasetModel = datasetService.retrieveModel(dataset, userReq);
```

In `DatasetSchemaUpdateResponseStep.doStep()` I also see:
```
    Dataset dataset = datasetDao.retrieve(datasetId);
    DatasetModel updatedDataset =
        datasetService.retrieveModel(
            dataset, userRequest, List.of(DatasetRequestAccessIncludeModel.SCHEMA));
```

Reply:
Nice, I was able to consolidate.